CMakeLists.txt

```diff
@@ -1,48 +1,51 @@
-cmake_minimum_required(VERSION 3.7...3.29)
-
-if (${CMAKE_VERSION} VERSION_LESS 3.12)
-    cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
-endif ()
+cmake_minimum_required(VERSION 3.23...3.29)
 
 project(WaraPSClient VERSION 1.0
         DESCRIPTION "WaraPS Client"
         LANGUAGES CXX
 )
 
+set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-add_library(WaraPSClient src/wara_ps_client.cc)
-
 find_package(PahoMqttCpp REQUIRED)
 find_package(nlohmann_json REQUIRED)
 
-target_link_libraries(WaraPSClient
-        PUBLIC PahoMqttCpp::paho-mqttpp3
-        PUBLIC nlohmann_json::nlohmann_json
+add_library(WaraPSClient)
+target_sources(WaraPSClient
+        PRIVATE
+        src/wara_ps_client.cc
+
+        PUBLIC
+        FILE_SET api
+        TYPE HEADERS
+        BASE_DIRS src
+        FILES src/wara_ps_client.h
+)
+target_link_libraries(WaraPSClient PUBLIC
+        PahoMqttCpp::paho-mqttpp3
+        nlohmann_json::nlohmann_json
 )
 
 include(GNUInstallDirs)
 
 install(TARGETS WaraPSClient
         EXPORT WaraPSClientTargets
-        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
-        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
-        INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+        FILE_SET api
 )
 
 install(EXPORT WaraPSClientTargets
         FILE WaraPSClientTargets.cmake
         NAMESPACE WaraPSClient::
-        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/WaraPSClient
```

Comment:
Problem likely that the cmake/WaraPSClient is in the `lib` directory in stead of in being in the root.